includes/Core/Modules/Module.php

```diff
@@ -343,7 +343,7 @@ final protected function execute_data_request( Data_Request $data ) {
 			// even if a different client will be the one to execute the request because
 			// the default instance is what services are setup with.
 			$restore_defers[] = $this->get_client()->withDefer( true );
-			if ( $this->get_client() !== $oauth_client ) {
+			if ( $this->authentication->get_oauth_client() !== $oauth_client ) {
```

Comment:
The purpose of this condition is to avoid stacking the defer; each callback in the array is intended to reference a restore for a different client.

This logical flaw resulted in a condition that would always be true, and so non-shared datapoint requests would be restored improperly. This broke the request in the `Analytics::parse_data_response` method which was expected to run without deferral.

Reply:
This makes me think, we may wanna do away with that `get_client()` method completely and instead just have a method like `get_oauth_client_for_current_user()`. That would fit more nicely with `get_oauth_client_for_datapoint()` in that both would return an `OAuth_Client_Base` instance rather than one returning a `Google_Site_Kit_Client` instance which is completely different.

Reply:
I filed #5368 for this change and left it in AC for now, but I like that idea 👍🏻